src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -2324,31 +2351,65 @@ public int GetHashCode(IAssemblySymbol? obj)
+            => oldProperty.ReturnsByRef == newProperty.ReturnsByRef &&
+               oldProperty.ReturnsByRefReadonly == newProperty.ReturnsByRefReadonly &&
+               oldProperty.TypeCustomModifiers.SequenceEqual(newProperty.TypeCustomModifiers, exact, (x, y, exact) => CustomModifiersEquivalent(x, y, exact)) &&
+               oldProperty.RefCustomModifiers.SequenceEqual(newProperty.RefCustomModifiers, exact, (x, y, exact) => CustomModifiersEquivalent(x, y, exact)) &&
+               TypesEquivalent(oldProperty.Type, newProperty.Type, exact);
+
+        protected static bool ReturnTypesEquivalent(IEventSymbol oldEvent, IEventSymbol newEvent, bool exact)
+            => TypesEquivalent(oldEvent.Type, newEvent.Type, exact);
+
+        // Note: SignatureTypeEquivalenceComparer compares dynamic and object the same.
+        protected static bool TypesEquivalent(ITypeSymbol? oldType, ITypeSymbol? newType, bool exact)
+            => (exact ? s_exactSymbolEqualityComparer : (IEqualityComparer<ITypeSymbol?>)s_runtimeSymbolEqualityComparer.SignatureTypeEquivalenceComparer).Equals(oldType, newType);
+
+        protected static bool ParameterTypesEquivalent(IParameterSymbol oldParameter, IParameterSymbol newParameter, bool exact)
+            => (exact ? s_exactSymbolEqualityComparer : s_runtimeSymbolEqualityComparer).ParameterEquivalenceComparer.Equals(oldParameter, newParameter);
+
+        protected static bool TypeParameterConstraintsEquivalent(ITypeParameterSymbol oldParameter, ITypeParameterSymbol newParameter, bool exact)
+            => oldParameter.ConstraintTypes.SequenceEqual(newParameter.ConstraintTypes, exact, (x, y, exact) => TypesEquivalent(x, y, exact)) &&
+               oldParameter.HasReferenceTypeConstraint == newParameter.HasReferenceTypeConstraint &&
+               oldParameter.HasValueTypeConstraint == newParameter.HasValueTypeConstraint &&
+               oldParameter.HasConstructorConstraint == newParameter.HasConstructorConstraint &&
+               oldParameter.HasNotNullConstraint == newParameter.HasNotNullConstraint &&
+               oldParameter.HasUnmanagedTypeConstraint == newParameter.HasUnmanagedTypeConstraint &&
+               oldParameter.Variance == newParameter.Variance;
+
+        protected static bool TypeParametersEquivalent(ImmutableArray<ITypeParameterSymbol> oldParameters, ImmutableArray<ITypeParameterSymbol> newParameters, bool exact)
+            => oldParameters.SequenceEqual(newParameters, exact, (oldParameter, newParameter, exact) => oldParameter.Name == newParameter.Name && TypeParameterConstraintsEquivalent(oldParameter, newParameter, exact));
+
+        protected static bool BaseTypesEquivalent(INamedTypeSymbol oldType, INamedTypeSymbol newType, bool exact)
+            => TypesEquivalent(oldType.BaseType, newType.BaseType, exact) &&
+               oldType.Interfaces.SequenceEqual(newType.Interfaces, exact, (x, y, exact) => TypesEquivalent(x, y, exact));
```

Comment:
Should this be AllInterfaces?
eg:
```
public interface B { }
public interface A : B { }

// source 1
public class C : A, B { }

// source2
public class C : A
```

Interfaces would be different, but AllInterfaces wouldn't be.

Reply:
Good point.